opium/src/middlewares/middleware_basic_auth.ml

```diff
@@ -0,0 +1,30 @@
+let m ?unauthorized_handler ~key ~realm ~auth_callback () =
+  let unauthorized_handler =
+    Option.value unauthorized_handler ~default:(fun _req ->
+        Response.of_plain_text "Forbidden access" ~status:`Unauthorized
+        |> Response.add_header ("WWW-Authenticate", Auth.string_of_challenge (Basic realm))
+        |> Lwt.return)
+  in
+  let filter handler ({ Request.env; _ } as req) =
+    let open Lwt.Syntax in
+    let+ resp =
+      match Request.authorization req with
+      | None -> unauthorized_handler req
+      | Some (Other _) -> handler req
```

Comment:
Should we handle other authentication methods this way? We have effectively not authenticated the user but still continue. The following curl snippet will avoid the `unauthorized_handler`, but in the example it a `Not_found` exception is raised because the user key is not set.

```
curl localhost:3000 -H "Authorization: Basic T0NhbWwK" # base64 for "OCaml" - no ':'
```

I think `unauthorized_handler` should be called setting bad request status or so.

Reply:
Do you mean handling everything with `handler req`, or handling other auth methods with `unauthorized_handler req`?

Reply:
Uh, I'm a bit confused by the question - but I'm happy with the change in the latest commit, so I consider this resolved.

Reply:
Ok great, I was confused by the opening question as well, but I see we had the same design in mind 🙂 
`Some (Other _) -> handler req` was an oversight